### ISSUE TITLE
Changelog on addons updates

### ISSUE THREAD

**Issue by javidominguez** [nvaccess/nvda#14041]
### Is your feature request related to a problem? Please describe.
Often when addon developers make changes in an update, many users are not even aware of it. Users usually read the documentation when they install a new addon but not every time they update it. Also, it would be tedious to read the entire documentation every time. Change information may be available on Github, on the addons website or on the mailing lists but not all users access it, maybe because they update via addonUpdater.

### Describe the solution you'd like
- Developers could include a changelog.md file in the doc folder of the addon.
- set a button in the addon manager, next to the "addon help" button, to show the changelog file.
- Set a button in the addon install dialog, when you are informed that there is already a version installed,to view the change log before updating it.

### Describe alternatives you've considered
- Any other way you can think of to display the change log in an easily accessible way for the user.

**Comment by seanbudd** [comment]
Blocked by #13985

**Comment by Cleversn** [comment]
I want this feature too. Each time I receive an add-on update warning, I'm forced to go look for the changes, usually at the add-ons discussion group to see if a thread was opened detailing the changes made.

**Comment by josephsl** [comment]
Hi,

2025 update: as a start, I think the add-on store submission form should have an edit field for recording changelog for the given release, which can double as a place to introduce new add-ons (if this is such a case).

Thanks.

**Comment by seanbudd** [comment]
I think this might be better off added to the manifest spec as an optional item

**Comment by josephsl** [comment]
Hi,

This means add-on authors must edit manifest or add-on template's buildVars (included in their add-on repos) whenever releases are made. The benefit is that this text can be made translatable, but it can take days for translators to localize the changelog text in the manifest. We also need to instruct add-on authors to never forget to write something in the changelog (and commit it) that can be understood by end users and easy to translate (I do have a changelog in my add-ons (included in the readme or a dedicated document is used) that can be committed to the changelog manifest key just before declaring a release (and tagging the release commit on Git).

While not easily translatable, a form field can accommodate folks who are not used to the add-on template approach (yet) or are pushing frequent add-on updates. But in the long term, a changelog manifest key would be a better approach (and can be useful thing to include in the eventual web interface to the NV Access add-on store).

Thanks.

**Comment by seanbudd** [comment]
Authors must edit these anyway for each release to update the version number

**Comment by josephsl** [comment]
Hi,

August 2025 update 2: a new manifest key, "changelog" will be added to the add-on template (template key: "addon_changelog"). The new manifest key will be a multi-line field describing the changelog for the version being released/tagged/published and should be discoverable from the add-on store view models.

Tanks.

**Comment by seanbudd** [comment]
@josephsl - would you consider making a PR to NVDA adding the manifest item? We don't support using custom manifest items in the add-on submission process, only official manifest entries, for example the custom "channel" attribute the community template uses.

**Comment by josephsl** [comment]
Hi,

Adding the manifest to add-on class, that I can try for 2025.3. As for the rest such as UI, I'll need to pass that on due to life priorities.

Also, as you may have gathered from the add-ons list content, the biggest concern is getting add-on authors to write changelogs for add-on releases (only a few of us do that at the moment).

Thanks.

**Comment by nvdaes** [comment]
@josephsl , @SaschaCowley , should this issue be reopened? I think this is not complete since changelog is not visible.
I mentioned, since Joseph said that he woldn't work in the gui for this feature, that I would do it, mentioning Joseph and I in NVDA's changelog.
I think that the changelog may be presented in the details pane, or below the description, which perhaps would be more visible and comfortable. Another alternative would be to create another readonly edit box for the changelog, but I think that the better approach is to show it below the description, which is supposed to be short according to the developer guide.
If this is not reopened, I can create a new issue, but I think that this should be included here.

**Comment by nvdaes** [comment]
Thinking better, I believe that it would be easier for users to add a different pane which could be found pressing tab after the description control, since some authors may write long descriptions, and many users may be interested just in the changelog, not in the add-on description.
I plan to work for this in a few hours.
